azure-pipelines.yml

```diff
@@ -63,32 +65,27 @@ stages:
             version: 'latest'
 
         - task: PowerShell@2
-          condition: eq(variables['Build.SourceBranch'], 'refs/release')
+          condition: eq(variables['Build.SourceBranch'], $(releaseBranch))
           displayName: 'Pre-release versioning - pubspec.yaml'
           inputs:
             targetType: 'inline'
             script: |
-              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern "(?<=version: ).*" | %{ $_.Matches[0].Value }
+              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern "$(versionGetRegex)" | %{ $_.Matches[0].Value }
               Write-Host "##vso[task.setvariable variable=currentVersion]$version"
 
-              [string] $versionRegex = '(?<=version: ).*'
               [string] $pubspecContent = Get-Content -Path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -Raw
 
               Write-Host " (i) Current version: $version"
               Write-Host " (i) Original pubspec.yaml content: $pubspecContent"
-
-              [bool] $hasMatches = $pubspecContent -match $versionRegex
-
-              Write-Host " (i) Found matches to replace version? $hasMatches"
               
-              [string] $newPubspecContent = $pubspecContent -replace $versionRegex, "$version-dev.$(Build.BuildID)"
+              [string] $newPubspecContent = $pubspecContent -replace $version, "$version-dev.$(Build.BuildID)"
```

Comment:
I think this shouldn't be changed?

Reply:
Either way works as both variable returns the same value, are you happy if I leave it as it is?

Reply:
Hmm, are we using regex in other places with similar code? If so I would stick to regex for consistency.

Reply:
Nay, that was the only line it uses `versionRegex` for a replace, so I'll leave it as it is